cmd/gateway-handlers.go

```diff
@@ -918,7 +923,9 @@ func (api gatewayAPIHandlers) GetBucketLocationHandler(w http.ResponseWriter, r
 		getBucketInfo = objectAPI.AnonGetBucketInfo
 	}
 
-	if _, err := getBucketInfo(bucket); err != nil {
+	_, err := getBucketInfo(bucket)
+	_, backendDown := errorCause(err).(BackendDown)
```

Comment:
how about doing a switch case? 
```
switch errorCause(err).(type) {
case BackendDown:
default:
```

Reply:
switch will not be very useful here - both backendDown and err == nil needs to be handled identically.